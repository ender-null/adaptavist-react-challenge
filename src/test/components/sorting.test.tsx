import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Sorting, { SortingProps, SortingTypes } from "../../components/sorting";

describe("Sorting", () => {
  const props: SortingProps = {
    currentSorting: "alphabetical",
    changeSorting: (sort: SortingTypes) => {
      return sort;
    },
  };
  test("should render the component", () => {
    const { asFragment } = render(<Sorting {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should run change the sorting to 'alphabetical' when clicked", () => {
    const { container } = render(
      <Sorting {...props} currentSorting="alphabetical" />
    );
    const node = container.querySelector('[key="alphabetical"]');
    if (node) {
      fireEvent.click(node);
      expect(props.changeSorting).toBe("alphabetical");
    } else {
      expect(container.firstChild).toBeFalsy;
    }
  });

  test("should run change the sorting to 'numerical' when clicked", () => {
    const { container } = render(
      <Sorting {...props} currentSorting="numerical" />
    );
    const node = container.querySelector('[key="numerical"]');
    if (node) {
      fireEvent.click(node);
      expect(props.changeSorting).toBe("numerical");
    } else {
      expect(container.firstChild).toBeFalsy;
    }
  });
});
