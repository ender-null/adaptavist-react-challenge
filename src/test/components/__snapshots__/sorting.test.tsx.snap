// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting should render the component 1`] = `
<DocumentFragment>
  <div
    class="sorting"
  >
    <button
      class="selected"
    >
      Alphabetical
    </button>
    <button
      class="unselected"
    >
      Numerical
    </button>
  </div>
</DocumentFragment>
`;
